client/my-sites/plans/main.jsx

```diff
@@ -36,7 +36,7 @@ const Plans = React.createClass( {
 	getDefaultProps() {
 		return {
 			intervalType: 'yearly',
-			displayJetpackPlans: true
+			displayJetpackPlans: false
```

Comment:
>  they need to specify it to true and will quickly notice that it shows wpcom plans if they don't (that's why I think it's safe).

well... my point is that is too easy to not notice if you are not paying attention: Same names, same icons, same number of plans ... if you don't stop to read the features, you won't even realize that the plans that you see are not jetpack plans. Heck, even the plan slug look similar :) 
It took me a while the other day to understand what was going on, I didn't notice that I was seeing wpcom plans until I was already debugging the code and I realized the plans didn't have a monthly version. 
So since the result of overlooking the parameter is pretty devastating, and the error may not be immediately noticeable ... I would prefer to make it much more explicit just throwing an exception if someone forgets to initialize it.